src/components/ProductMilestoneInfoList/ProductMilestoneInfoList.tsx

```diff
@@ -0,0 +1,119 @@
+import { Link } from 'react-router-dom';
+
+import { MilestoneInfo } from 'pnc-api-types-ts';
+
+import { PageTitles } from 'common/constants';
+import { getFilterAttributes } from 'common/entityAttributes';
+import { milestoneInfoEntityAttributes } from 'common/milestoneInfoEntityAttributes';
+
+import { IServiceContainer } from 'hooks/useServiceContainer';
+import { ISortAttributes, useSorting } from 'hooks/useSorting';
+
+import { ContentBox } from 'components/ContentBox/ContentBox';
+import { Filtering } from 'components/Filtering/Filtering';
+import { Pagination } from 'components/Pagination/Pagination';
+import { ServiceContainerLoading } from 'components/ServiceContainers/ServiceContainerLoading';
+import { Toolbar } from 'components/Toolbar/Toolbar';
+import { ToolbarItem } from 'components/Toolbar/ToolbarItem';
+
+import { createDateTime } from 'utils/utils';
+
+const sortAttributes: ISortAttributes = {
+  milestoneEndDate: {
+    id: 'milestoneEndDate',
+    title: 'Completed On',
+    tableColumnIndex: 3,
+  },
+  releaseReleaseDate: {
+    id: 'releaseReleaseDate',
+    title: 'Released On',
+    tableColumnIndex: 5,
+  },
+};
+
+interface IProductMilestoneInfoListProps {
+  serviceContainerProductMilestoneInfo: IServiceContainer;
+  componentId: string;
+}
+
+/**
+ * Component displaying list of Product Milestones that produced or consumed an Artifact.
+ *
+ * @param serviceContainerProductMilestoneInfo - Service Container for Product Milestone info
+ * @param componentId - Component ID
+ */
+export const ProductMilestoneInfoList = ({
+  serviceContainerProductMilestoneInfo,
+  componentId,
+}: IProductMilestoneInfoListProps) => {
+  const { getSortParams } = useSorting(sortAttributes, componentId);
+
+  return (
+    <>
+      <Toolbar>
+        <ToolbarItem>
+          <Filtering filterOptions={getFilterAttributes(milestoneInfoEntityAttributes)} componentId={componentId} />
+        </ToolbarItem>
+      </Toolbar>
+
+      <ContentBox borderTop>
+        <ServiceContainerLoading {...serviceContainerProductMilestoneInfo} title={PageTitles.productMilestones}>
+          <TableComposable isStriped variant="compact">
+            <Thead>
+              <Tr>
+                <Th width={20}>{milestoneInfoEntityAttributes.productName.title}</Th>
+                <Th width={15}>{milestoneInfoEntityAttributes.productVersionVersion.title}</Th>
+                <Th width={15}>{milestoneInfoEntityAttributes.milestoneVersion.title}</Th>
+                <Th
+                  info={{
+                    tooltip: milestoneInfoEntityAttributes.milestoneEndDate.tooltip,
+                  }}
+                  sort={getSortParams(sortAttributes['milestoneEndDate'].id)}
+                >
+                  {milestoneInfoEntityAttributes.milestoneEndDate.title}
+                </Th>
+                <Th width={15}>{milestoneInfoEntityAttributes.releaseVersion.title}</Th>
```

Comment:
I would use `Release {productReleaseEntityAttributes.version.title}`.

Apply the same on other places too.

Reply:
done (corrected version)